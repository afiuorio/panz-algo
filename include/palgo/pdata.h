#ifndef PDATA_H_
#define PDATA_H_

void **alloc_pMatrix(size_t numRow, size_t numCol, size_t elem_size);
void free_pMatrix(void **matrix);

#endif
