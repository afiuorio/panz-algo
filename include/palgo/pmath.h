#ifndef PMATH_H_
#define PMATH_H_

int p_abs(int n);

double p_cos(double x);
double p_sin(double x);

double p_sqrt(double n);

double p_pow(double x, double n);

#endif
