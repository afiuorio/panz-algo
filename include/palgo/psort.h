#ifndef PSORT_H_
#define PSORT_H_

void p_sort(void *base, size_t n, size_t size, int(*cmp)(const void *,const void *));

void p_isort(void *base, size_t n, size_t size,int(*cmp)(const void*,const void*));

int p_rsort(unsigned int *array, size_t n);

#endif
