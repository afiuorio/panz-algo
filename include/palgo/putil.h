#ifndef PUTIL_H_
#define PUTIL_H_

#define PRAND_MAX 0x7fff

void p_srand(unsigned int seed);
int p_rand();

void *p_search(void *key,void *array, size_t num ,size_t size,int(*cmp)(const void *,const void *) );

char *p_strcpy(char *s, const char *ct);
char *p_strcat(char *s, const char *ct);
int p_strcmp(const char *cs, const char *ct);

#endif
