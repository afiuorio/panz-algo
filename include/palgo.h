#ifndef PALGO_H_
#define PALGO_H_

#include "palgo/pdata.h"
#include "palgo/pmath.h"
#include "palgo/psort.h"
#include "palgo/putil.h"

#endif
