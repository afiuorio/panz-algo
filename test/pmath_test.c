#include<stdio.h>
#include<palgo.h>
#include<math.h>

void test_sqrt(){
	printf("%.13f\n", p_sqrt(2));
	printf("%.13f\n", p_sqrt(3.14));
	printf("%.13f\n", p_sqrt(25));
	printf("%.13f\n", p_sqrt(196));
	printf("%.13f\n", p_sqrt(2000));
}

void test_cordic(){
	printf("%.13f\n", p_cos(2));
	printf("%.13f\n", p_cos(3.14159265358979323846));
	printf("%.13f\n", p_cos(1.5));
	printf("%.13f\n", p_cos(3* 3.14159265358979323846));
	printf("%.13f\n", p_sin(2));
	//printf("%.16f\n", p_sin(3.14159265358979323846));
	printf("%.13f\n", p_sin(1.5));
	//printf("%.16f\n", p_sin(3* 3.14159265358979323846));
}

int main(){
	test_cordic();
	test_sqrt();
}
