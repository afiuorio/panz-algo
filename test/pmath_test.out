-0.4161468365471
-1.0000000000000
0.0707372016677
-1.0000000000000
0.9092974268257
0.9974949866041
1.4142135623731
1.7720045146669
5.0000000000000
14.0000000000000
44.7213595499958
